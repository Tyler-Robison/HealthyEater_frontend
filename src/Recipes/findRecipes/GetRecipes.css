.GetRecipes {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-image: url('../../Images/food_background.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    background-attachment: fixed;
    /* height: 92.9vh; */
}

.GetRecipes-explainer-div {
    margin: 1.5rem 0px;
    /* width: 26vw; */
    background-color: white;
    padding: 1rem 0rem;;
    border-radius: 1rem;
    box-shadow: 2px 2px 8px rgba(0, 0, 0, 1);
} 

@media (min-width: 400px) {
    .GetRecipes-explainer-div {
        font-size: 1.2rem;
    }
}

@media (max-width: 400px) {
    .GetRecipes-explainer-div {
        font-size: 0.9rem;
    }
}

.GetRecipes-explainer-para {
    margin: 0px;
    font-weight: 700;
    /* font-size: 1.2rem; */
}

.GetRecipes-container {
    border: solid 2px black;
    padding: 0;
    margin: 0;
    /* width: 100vw; */
    /* height: 60vh; */
    /* display: flex;
    flex-direction: row; */
}

.GetRecipes-ingredient-div {
    background-image: url('../../Images/food_theme.jpg');
    /* background-image */
    /* width: 50vw; */
    display: flex;
    flex-direction: column;
    align-items: center;

}

.GetRecipes-nutrient-div {
    background-image: url('../../Images/nutri_background.jpg');
    /* width: 49.9vw; */
    display: flex;
    flex-direction: column;
    align-items: center;
}


.GetRecipes-btn {
    margin: 1rem 0;
}

.GetRecipes-header {
    margin-top: 1rem;
    color: #2477f3;
    font-weight: 900;
    font-size: 2.2rem;
}

.GetRecipes-IngForm-container {
    background-color: white;
}

.GetRecipes-NutForm-container {
    background-color: rgb(191, 227, 240);
}

.content-box {
    min-width: 320px;
    max-width: 400px;
    margin: 1rem 0;
    border-radius: 1rem;
    box-shadow: 2px 2px 8px rgba(0, 0, 0, 1); 
    height: 100%;
    padding: 0 1rem;
}



