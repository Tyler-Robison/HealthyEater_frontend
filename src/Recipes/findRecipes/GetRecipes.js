import React, { useContext, useState } from "react";
import SpoonacularAPI from "../../APIs/spoonAPI";
import IngredientForm from "./IngredientForm";
import NutrientForm from "./NutrientForm";
import IngredientList from "./IngredientList";
// import useLocalStorage from "../../customHooks/useLocalStorage";
import Message from "../../TopLevel/Message";
import GlobalContext from "../../context/GlobalContext";
import './GetRecipes.css'
import { useNavigate } from "react-router-dom";

const GetRecipes = () => {
    const { setRecipes, setMsg, msg, token, currentUser } = useContext(GlobalContext)

    const navigate = useNavigate();
    const [ingredientsList, setIngredientsList] = useState([])

    const [fat, setFat] = useState(null)
    const [satFat, setSatFat] = useState(null)
    const [calories, setCalories] = useState(null)
    const [carbs, setCarbs] = useState(null)
    const [sugar, setSugar] = useState(null)
    const [sodium, setSodium] = useState(null)
    const [cholesterol, setCholesterol] = useState(null)
    const [protein, setProtein] = useState(null)

    // contains min/max values all recipes must fall within
    const nutrientObj = {
        maxFat: fat,
        maxSaturatedFat: satFat,
        maxCalories: calories,
        maxCarbs: carbs,
        maxCholesterol: cholesterol,
        maxSugar: sugar,
        maxSodium: sodium,
        maxProtein: protein
    }

    const getRecipesFunc = async () => {
        const res = await SpoonacularAPI.getRecipes(ingredientsList, nutrientObj, token, currentUser.id)
        console.log('recipe res', res)
        if (res.results.length === 0) {
            setMsg('Recipe search failed. Try removing ingredients. If you have nutritional constraints, try relaxing them.')
            return
        } else {
            setMsg('')
        }
        setRecipes(res.results);
        navigate('/recipes');
    }

    return (
        <div className="container-div">
            <div className="FindRecipes-div">
                <h2>Recipe Search</h2>
                <p className="FindRecipes-para">1) Enter ingredients one at a time</p>
                <p className="FindRecipes-para">2) Optionally, add nutrient requirements</p>
                <p className="FindRecipes-para">3) Click "Get Recipes"</p>
            </div>
            <div className="recipe-div">
                <div className="ingredient-div">
                    {/* both keys can be id not index */}

                    <div className="GetRecipes-IngForm-container">
                        <h3 className="GetRecipes-header">Ingredients</h3>
                        <Message msg={msg} />
                        <IngredientForm ingredientsList={ingredientsList} setIngredientsList={setIngredientsList} />
                        <IngredientList ingredientsList={ingredientsList} setIngredientsList={setIngredientsList} />
                        <button className="general-btn GetRecipes-btn" onClick={getRecipesFunc}>Get Recipes</button>
                    </div>
                </div>
                <div className="nutrient-div">
                    <div className="NutrientForm-container">
                        <h3 className="GetRecipes-header">Nutrients</h3>
                        {/* <button onClick={resetNutrients}>Clear List</button> */}

                        <NutrientForm
                            setFat={setFat}
                            setSatFat={setSatFat}
                            setSugar={setSugar}
                            setSodium={setSodium}
                            setCholesterol={setCholesterol}
                            setCalories={setCalories}
                            setCarbs={setCarbs}
                            setProtein={setProtein}
                        />
                    </div>
                </div>
            </div>
        </div>
    )
}

export default GetRecipes;