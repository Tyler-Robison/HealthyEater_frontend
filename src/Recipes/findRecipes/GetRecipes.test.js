import { render, screen } from '@testing-library/react';
import GetRecipes from './GetRecipes';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext'


it("renders without crashing", function () {
    <MemoryRouter>
        <GetRecipes />
    </MemoryRouter>
});


it("matches snapshot", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <GetRecipes />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});