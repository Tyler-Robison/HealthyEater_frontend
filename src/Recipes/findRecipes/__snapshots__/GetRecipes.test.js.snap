// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-div"
  >
    <div
      class="FindRecipes-div"
    >
      <h2>
        Recipe Search
      </h2>
      <p
        class="FindRecipes-para"
      >
        1) Enter ingredients one at a time
      </p>
      <p
        class="FindRecipes-para"
      >
        2) Optionally, add nutrient requirements
      </p>
      <p
        class="FindRecipes-para"
      >
        3) Click "Get Recipes"
      </p>
    </div>
    <div
      class="recipe-div"
    >
      <div
        class="ingredient-div"
      >
        <div
          class="GetRecipes-IngForm-container"
        >
          <h3
            class="GetRecipes-header"
          >
            Ingredients
          </h3>
          <h3 />
          <form
            class="AddIngredient-Form"
          >
            <input
              class="test-input"
              id="ingredient"
              name="ingredient"
              placeholder="Enter Ingredient"
              role=""
              type="text"
              value=""
            />
            <button
              class="general-btn"
              type="submit"
            >
              Add Ingredient
            </button>
          </form>
          <div />
          <button
            class="general-btn GetRecipes-btn"
          >
            Get Recipes
          </button>
        </div>
      </div>
      <div
        class="nutrient-div"
      >
        <div
          class="NutrientForm-container"
        >
          <h3
            class="GetRecipes-header"
          >
            Nutrients
          </h3>
          <form
            class="AddIngredient-Form"
          >
            <div
              class="nutrient-form-div"
            >
              <div
                class="nutrient-input-div"
              >
                <div>
                  <label
                    for="fat"
                  >
                    Max Fat (grams)
                  </label>
                </div>
                <input
                  id="fat"
                  name="maxFat"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="nutrient-input-div"
              >
                <div>
                  <label
                    for="satFat"
                  >
                    Max Sat Fat (grams)
                  </label>
                </div>
                <input
                  id="satFat"
                  name="maxSaturatedFat"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="nutrient-input-div"
              >
                <div>
                  <label
                    for="carbs"
                  >
                    Max Carbs (grams)
                  </label>
                </div>
                <input
                  id="carbs"
                  name="maxCarbs"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="nutrient-input-div"
              >
                <div>
                  <label
                    for="calories"
                  >
                     Max Calories (kCals)
                  </label>
                </div>
                <input
                  id="calories"
                  name="maxCalories"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="nutrient-input-div"
              >
                <div>
                  <label
                    for="cholesterol"
                  >
                    Max Cholesterol (mg)
                  </label>
                </div>
                <input
                  id="cholesterol"
                  name="maxCholesterol"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="nutrient-input-div"
              >
                <div>
                  <label
                    for="sugar"
                  >
                    Max Sugar (grams)
                  </label>
                </div>
                <input
                  id="sugar"
                  name="maxSugar"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="nutrient-input-div"
              >
                <div>
                  <label
                    for="sodium"
                  >
                    Max Sodium (mg)
                  </label>
                </div>
                <input
                  id="sodium"
                  name="maxSodium"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="nutrient-input-div"
              >
                <div>
                  <label
                    for="protein"
                  >
                    Min Protein (grams)
                  </label>
                </div>
                <input
                  id="protein"
                  name="minProtein"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div>
              <button
                class="general-btn set-nutrients-btn"
                type="submit"
              >
                Update Values
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
